lib/net.js

```diff
@@ -1755,7 +1752,7 @@ function Server(options, connectionListener) {
   this.noDelay = Boolean(options.noDelay);
   this.keepAlive = Boolean(options.keepAlive);
   this.keepAliveInitialDelay = ~~(options.keepAliveInitialDelay / 1000);
-  this.highWaterMark = options.highWaterMark ?? getDefaultHighWaterMark();
+  this.highWaterMark = options.highWaterMark ?? undefined;
```

Comment:
Theoretically `this.highWaterMark = options.highWaterMark;` should suffice if we are happy with falsy values.

Reply:
Yes, let's accept falsy values.